java/kotlin-extractor2/src/main/kotlin/entities/Expression.kt

```diff
@@ -311,57 +319,65 @@ private fun KotlinFileExtractor.extractExpression(
-                                tw.writeHasLocation(id, locId)
-                                extractExpressionStmt(e.tryResult, callable, id, -1)
-                                val finallyStmt = e.finallyExpression
-                                if (finallyStmt != null) {
-                                    extractExpressionStmt(finallyStmt, callable, id, -2)
-                                }
-                                for ((catchIdx, catchClause) in e.catches.withIndex()) {
-                                    val catchId = tw.getFreshIdLabel<DbCatchclause>()
-                                    tw.writeStmts_catchclause(catchId, id, catchIdx, callable)
-                                    val catchLocId = tw.getLocation(catchClause)
-                                    tw.writeHasLocation(catchId, catchLocId)
-                                    extractTypeAccessRecursive(
-                                        catchClause.catchParameter.type,
-                                        tw.getLocation(catchClause.catchParameter),
-                                        catchId,
-                                        -1,
-                                        callable,
-                                        catchId
-                                    )
-                                    extractVariableExpr(
-                                        catchClause.catchParameter,
-                                        callable,
-                                        catchId,
-                                        0,
-                                        catchId
-                                    )
-                                    extractExpressionStmt(catchClause.result, callable, catchId, 1)
-                                }
-                            }
-                            is IrContainerExpression -> {
-                                if (
-                                    !tryExtractArrayUpdate(e, callable, parent) &&
-                                        !tryExtractForLoop(e, callable, parent)
-                                ) {
+            is KtTryExpression -> {
+                val stmtParent = parent.stmt(e, callable)
+                val id = tw.getFreshIdLabel<DbTrystmt>()
+                val locId = tw.getLocation(e)
+                tw.writeStmts_trystmt(id, stmtParent.parent, stmtParent.idx, callable)
+                tw.writeHasLocation(id, locId)
+                extractExpressionStmt(e.tryBlock, callable, id, -1)
+                val finallyStmt = e.finallyBlock?.finalExpression
+                if (finallyStmt != null) {
+                    extractExpressionStmt(finallyStmt, callable, id, -2)
+                }
+                for ((catchIdx, catchClause) in e.catchClauses.withIndex()) {
+                    val catchId = tw.getFreshIdLabel<DbCatchclause>()
+                    tw.writeStmts_catchclause(catchId, id, catchIdx, callable)
```

Comment:
It seems a little odd that the dbscheme has a `catchclause` but not a `finallyclause`. I wonder if we should change that.